src/lib/transform.js

```diff
@@ -193,16 +193,31 @@ function makeGoalsAndObjectivesObject(objectiveRecords) {
   objectiveRecords.sort(sortObjectives);
   let objectiveNum = 0;
   let goalNum = 0;
+  const goalIds = {};
 
-  return objectiveRecords.reduce((accum, objective) => {
+  return objectiveRecords.reduce((prevAccum, objective) => {
+    const accum = { ...prevAccum };
     const {
-      goal, title, status, ttaProvided,
+      goal, title, status, ttaProvided, roles, topics, files, resources,
     } = objective;
+    const goalId = goal ? goal.id : null;
     const goalName = goal ? goal.name : null;
```

Comment:
Do we want to nope out here if there isn't an id or a name? Why wouldn't there be either of those?

Reply:
Im thinking maybe other entity otherwise just a test.

Reply:
objectives from other entities don't have goals.